runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractQueuedStreamView.java

```diff
@@ -400,9 +400,8 @@ protected boolean fillReadQueue(final long maxGlobal,
                     return true;
                 }
             } catch (TrimmedException te) {
-                // If we reached a trim and didn't hit a checkpoint, this might be okay,
-                // if the stream was created recently and no checkpoint exists yet.
-                log.warn("Fill_Read_Queue[{}] Trim encountered and no checkpoint detected.", this);
+                log.warn("Fill_Read_Queue[{}] Trim encountered.", this);
+                throw te;
```

Comment:
Can we add a comment here how throwing a TrimmedException here will later reset the object and fetch the latest checkpoint? Also if you want, maybe not in this patch, you can add a testcase later to test this situation to see if after TrimmedException, the latest checkpoint is correctly retrieved.